app/PaymentDrivers/Blockonomics/Blockonomics.php

```diff
@@ -144,41 +146,68 @@ public function paymentResponse(PaymentResponseRequest $request)
             // to prevent duplicate entries in the database.
-            // This ensures the payment hashed_id remains unique.
             $testTxid = $this->test_txid;
             $data['transaction_reference'] = ($request->txid === $testTxid)
                 ? $request->txid . bin2hex(random_bytes(16))
                 : $request->txid;
 
-            $statusId = Payment::STATUS_PENDING;
-
-            switch ($request->status) {
-                case 0:
-                    $statusId = Payment::STATUS_PENDING;
-                    break;
-                case 1:
-                    $statusId = Payment::STATUS_PENDING;
-                    break;
-                case 2:
-                    $statusId = Payment::STATUS_COMPLETED;
-                    break;
-                default:
-                    $statusId = Payment::STATUS_PENDING;
+            // Determine payment status
+            $statusId = match($request->status) {
+                2 => Payment::STATUS_COMPLETED,
+                default => Payment::STATUS_PENDING
+            };
+
+            // Get payment hash and invoice before creating payment
+            $payment_hash = PaymentHash::where('hash', $request->payment_hash)->firstOrFail();
+            $invoice = $payment_hash->fee_invoice ?? $payment_hash->paymentable;
+
+            if (!$invoice) {
+                throw new PaymentFailed('No invoice found for payment hash');
             }
 
+            // Store original invoice state for pending payments
+            $original_balance = $invoice->balance;
+            $original_status = $invoice->status_id;
+
+            // Create the payment - let InvoiceNinja handle the initial processing
             $payment = $this->blockonomics->createPayment($data, $statusId);
             $payment->private_notes = "{$request->btc_address} - {$request->btc_amount}";
             $payment->save();
 
+            // Alternative: Get invoice from payment after creation
+            $invoice = $invoice ?? $payment->invoices()->first();
+
+            // Refresh invoice to get InvoiceNinja's automatic updates
+            $invoice->refresh();
+
+            // Handle invoice status based on payment confirmation
+            if ($request->status == 2) {
+                // Payment is confirmed - let InvoiceNinja's natural flow work
+                // Just ensure status is correct based on balance
+                if ($invoice->balance <= 0) {
+                    $invoice->status_id = Invoice::STATUS_PAID;
+                } else {
+                    $invoice->status_id = Invoice::STATUS_PARTIAL;
+                }
+                $invoice->save();
+            } else {
+                // Payment is pending - revert InvoiceNinja's automatic changes
+                $invoice->status_id = $original_status ?: Invoice::STATUS_SENT;
+                $invoice->balance = $original_balance;
+                $invoice->save();
+            }
```

Comment:
This is not required.
The createPayment() will resolve attached invoices and update their balances.